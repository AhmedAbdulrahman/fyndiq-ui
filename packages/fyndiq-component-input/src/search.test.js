import React from 'react'
import { shallow, mount } from 'enzyme'

import Search from './search'

Math.random = jest.fn(() => 'RANDOM-ID')

describe('fyndiq-component-input SearchInput', () => {
  it('should have the proper structure', () => {
    expect(shallow(<Search />)).toMatchSnapshot()
  })

  it('should have a size prop', () => {
    expect(shallow(<Search size="s" />).hasClass('wrapper--size-s')).toBe(true)
  })

  it('should call onSearch', () => {
    const spy = jest.fn()
    const component = shallow(<Search onSearch={spy} />)
    component.find('Input').simulate('change', 'newValue')
    component.find('form').simulate('submit', { preventDefault: jest.fn() })

    expect(spy).toHaveBeenCalledWith('newValue')
  })

  it('should empty the search field if emptyAfterSearch is true', () => {
    const component = mount(<Search emptyAfterSearch />)
    component.find('Input').simulate('change', 'newValue')
    component.find('form').simulate('submit', { preventDefault: jest.fn() })

    expect(component.find('Input').prop('value')).toBe('')
  })

  it('should blur the input when pressing ESC', () => {
    const spy = jest.fn()
    const component = mount(<Search emptyAfterSearch />)
    component.getDOMNode().querySelector('input').blur = spy
    component.find('Input').simulate('keyup', { keyCode: 27 })
    expect(spy).toHaveBeenCalled()
  })

  it('should update its value if the props change', () => {
    const component = shallow(<Search value="test" />)
    component.setProps({ value: 'newValue' })
    expect(component.find('Input').prop('value')).toBe('newValue')
  })

  describe('collapsible prop', () => {
    it('should not be open by default', () => {
      expect(shallow(<Search collapsible />).hasClass('wrapperOpen')).toBe(
        false,
      )
    })

    it('should be open when focusing the input', () => {
      const component = shallow(<Search collapsible />)
      component.find('Input').simulate('focus')
      expect(component.hasClass('wrapperOpen')).toBe(true)
    })

    it('should be closed when bluring the input', () => {
      const component = shallow(<Search collapsible />)
      component.find('Input').simulate('focus')
      component.find('Input').simulate('blur')
      expect(component.hasClass('wrapperOpen')).toBe(false)
    })

    it('should always be open if not collapsible', () => {
      const component = shallow(<Search />)
      expect(component.hasClass('wrapperOpen')).toBe(true)

      component.find('Input').simulate('focus')
      expect(component.hasClass('wrapperOpen')).toBe(true)

      component.find('Input').simulate('blur')
      expect(component.hasClass('wrapperOpen')).toBe(true)
    })
  })
})
