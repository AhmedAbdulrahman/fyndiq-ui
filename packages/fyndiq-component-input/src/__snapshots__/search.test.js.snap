// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fyndiq-component-input SearchInput should have the proper structure 1`] = `
<form
  className="
          wrapper
          wrapperOpen
          wrapper--size-m
          
        "
  onSubmit={[Function]}
>
  <label
    className="label"
    htmlFor="RANDOM-ID"
  >
    <Magnifier
      className="magnifier"
    />
  </label>
  <Input
    autoComplete="off"
    className="input"
    debounceWait={500}
    debouncedOnChange={null}
    id="RANDOM-ID"
    inputRef={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyUp={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
</form>
`;
